corehq/apps/locations/fixtures.py

```diff
@@ -103,19 +109,26 @@ def _location_footprint(locations):
     queue = list(locations)
     while queue:
         loc = queue.pop()
-        assert loc._id in all_locs
-        if loc.parent_id and loc.parent_id not in all_locs:
-            all_locs.add_location(loc.parent)
-            queue.append(loc.parent)
+
+        if loc.location_id not in all_locs:
+            # if it's not in there, it wasn't valid
+            continue
+
+        parent = loc.parent
+        if (parent and
+                parent.location_id not in all_locs and
+                _valid_parent_type(loc)):
+            all_locs.add_location(parent)
+            queue.append(parent)
 
     return all_locs
 
 
-def _append_children(node, location_db, locations, type_lookup_function):
+def _append_children(node, location_db, locations):
     by_type = _group_by_type(locations)
     for type, locs in by_type.items():
         locs = sorted(locs, key=lambda loc: loc.name)
-        node.append(_types_to_fixture(location_db, type, locs, type_lookup_function))
+        node.append(_types_to_fixture(location_db, type, locs))
 
 
 def _group_by_type(locations):
```

Comment:
fun fact - this is using an object as a key in a dictionary - apparently the Django ORM supports that.